src/find-options/FindOptionsWhere.ts

```diff
@@ -5,34 +5,37 @@ import { EqualOperator } from "./EqualOperator"
 /**
  * A single property handler for FindOptionsWhere.
  */
-export type FindOptionsWhereProperty<Property> = Property extends Promise<
-    infer I
->
+export type FindOptionsWhereProperty<
+    Property,
+    Original = Property,
+> = Property extends Promise<infer I>
     ? FindOptionsWhereProperty<NonNullable<I>>
     : Property extends Array<infer I>
     ? FindOptionsWhereProperty<NonNullable<I>>
     : Property extends Function
     ? never
     : Property extends Buffer
-    ? Property | FindOperator<Property>
+    ? Property | FindOperator<Property> | FindOperator<Original>
```

Comment:
The problem is that the union is narrowed down when `extends` is used. So we keep the original Union as the `Original` and pass the generics to the `FindOperator` while keeping the Union.

Reply:
I wanted to make Distributive Conditional Types fix by adding square brakets instead of these changes, but I failed to make it to work.

Okay, one question I have - why do we need `FindOperator<Property> | FindOperator<Original>` and not just `FindOperator<Original>` now?